### ISSUE TITLE
ICE: `tried to get type of this RPITIT with no definition`

### ISSUE THREAD

**Issue by PhiSyX** [rust-lang/rust-clippy#11209]
### Summary

This code with feature `#![feature(return_position_impl_trait_in_trait)]` (`impl Trait`) causes an ICE error with the `cargo clippy` command.

```Rust
#![feature(return_position_impl_trait_in_trait)]

pub trait Application {
    fn setup() -> Result<(), impl std::error::Error>;
}
```

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=b12a32ed273c7e34b4bb13a855659f7e

### Version

```text
rustc 1.73.0-nightly (39f42ad9e 2023-07-19)
binary: rustc
commit-hash: 39f42ad9e8430a8abb06c262346e89593278c515
commit-date: 2023-07-19
host: x86_64-pc-windows-msvc
release: 1.73.0-nightly
LLVM version: 16.0.5
```

### Error output

```
Checking playground v0.0.1 (/playground)
error: internal compiler error: compiler/rustc_hir_analysis/src/collect/type_of.rs:448:25: tried to get type of this RPITIT with no definition
 --> src/lib.rs:4:30
  |
4 |     fn setup() -> Result<(), impl std::error::Error>;
  |                              ^^^^^^^^^^^^^^^^^^^^^^

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: please attach the file at `/playground/rustc-ice-2023-07-22T00:22:01.599597046Z-17.txt` to your bug report

note: compiler flags: --crate-type lib -C embed-bitcode=no -C codegen-units=1 -C debuginfo=2

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [type_of] computing type of `Application::setup::{opaque#0}`
#1 [normalize_projection_ty] normalizing `<Self as Application>::{opaque#0}`
#2 [try_normalize_generic_arg_after_erasing_regions] normalizing `<Self as Application>::{opaque#0}`
#3 [layout_of] computing layout of `<Self as Application>::{opaque#0}`
#4 [analysis] running analysis passes on this crate
end of query stack
note: Clippy version: clippy 0.1.73 (399b068 2023-07-20)

error: could not compile `playground` (lib)
```

<details><summary>Backtrace</summary>
  <p>

  ```
read 'rustc' panicked at 'Box<dyn Any>', /rustc/399b068235ceea440540539b3bfd1aeb82214a28/compiler/rustc_errors/src/lib.rs:1007:33
stack backtrace:
   0:     0x7f3d9ad55e71 - std::backtrace_rs::backtrace::libunwind::trace::h8fd05b5a530fa3a7
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f3d9ad55e71 - std::backtrace_rs::backtrace::trace_unsynchronized::h8a8d11ec8ab673c4
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f3d9ad55e71 - std::sys_common::backtrace::_print_fmt::h7d69e6adf84af8e3
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f3d9ad55e71 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h14edde268589a7dc
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f3d9adbc1fc - core::fmt::rt::Argument::fmt::had85f85f75a9e38f
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/core/src/fmt/rt.rs:138:9
   5:     0x7f3d9adbc1fc - core::fmt::write::h9d64dcd16df3df12
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f3d9ad4877e - std::io::Write::write_fmt::hea568404be2874a7
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/io/mod.rs:1714:15
   7:     0x7f3d9ad55c85 - std::sys_common::backtrace::_print::h76db4f3016b100e4
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f3d9ad55c85 - std::sys_common::backtrace::print::h40b955855ddd547c
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f3d9ad58cda - std::panicking::panic_hook_with_disk_dump::{{closure}}::h8ac97856201eec78
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/panicking.rs:278:22
  10:     0x7f3d9ad58973 - std::panicking::panic_hook_with_disk_dump::h0bf2ea8ad57ecce9
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/panicking.rs:312:9
  11:     0x7f3d9df88029 - <rustc_driver_impl[7a6d5a32edd3e2a]::install_ice_hook::{closure#0} as core[cd6e598b5651e7ef]::ops::function::FnOnce<(&core[cd6e598b5651e7ef]::panic::panic_info::PanicInfo,)>>::call_once::{shim:vtable#0}
  12:     0x7f3d9ad59580 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h29cdf48b8ba8b6ae
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/alloc/src/boxed.rs:2021:9
  13:     0x7f3d9ad59580 - std::panicking::rust_panic_with_hook::h63b949f8416b37cd
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/panicking.rs:733:13
  14:     0x7f3d9e0e0c51 - std[28b408c5a4d1cfeb]::panicking::begin_panic::<rustc_errors[dbd1c1d6b2fd1a6d]::ExplicitBug>::{closure#0}
  15:     0x7f3d9e0e0646 - std[28b408c5a4d1cfeb]::sys_common::backtrace::__rust_end_short_backtrace::<std[28b408c5a4d1cfeb]::panicking::begin_panic<rustc_errors[dbd1c1d6b2fd1a6d]::ExplicitBug>::{closure#0}, !>
  16:     0x7f3d9e12c3e6 - std[28b408c5a4d1cfeb]::panicking::begin_panic::<rustc_errors[dbd1c1d6b2fd1a6d]::ExplicitBug>
  17:     0x7f3d9e10c4ae - <rustc_errors[dbd1c1d6b2fd1a6d]::HandlerInner>::span_bug::<rustc_span[15bad65b23151bca]::span_encoding::Span, alloc[40e9e1a7dd990770]::string::String>
  18:     0x7f3d9e10c1df - <rustc_errors[dbd1c1d6b2fd1a6d]::Handler>::span_bug::<rustc_span[15bad65b23151bca]::span_encoding::Span, alloc[40e9e1a7dd990770]::string::String>
  19:     0x7f3d9e0d7fed - rustc_middle[ad940b573bf0e24a]::util::bug::opt_span_bug_fmt::<rustc_span[15bad65b23151bca]::span_encoding::Span>::{closure#0}
  20:     0x7f3d9e0d801a - rustc_middle[ad940b573bf0e24a]::ty::context::tls::with_opt::<rustc_middle[ad940b573bf0e24a]::util::bug::opt_span_bug_fmt<rustc_span[15bad65b23151bca]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  21:     0x7f3d9e0d1d48 - rustc_middle[ad940b573bf0e24a]::ty::context::tls::with_context_opt::<rustc_middle[ad940b573bf0e24a]::ty::context::tls::with_opt<rustc_middle[ad940b573bf0e24a]::util::bug::opt_span_bug_fmt<rustc_span[15bad65b23151bca]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  22:     0x7f3d9e0d1d24 - rustc_middle[ad940b573bf0e24a]::util::bug::span_bug_fmt::<rustc_span[15bad65b23151bca]::span_encoding::Span>
  23:     0x7f3d9d189b54 - rustc_hir_analysis[1ffe863f29b7852d]::collect::type_of::type_of
  24:     0x7f3d9c13b763 - rustc_query_impl[38c8fda5da4ce4ba]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[38c8fda5da4ce4ba]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>
  25:     0x7f3d9c13b720 - <rustc_query_impl[38c8fda5da4ce4ba]::query_impl::type_of::dynamic_query::{closure#2} as core[cd6e598b5651e7ef]::ops::function::FnOnce<(rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt, rustc_span[15bad65b23151bca]::def_id::DefId)>>::call_once
  26:     0x7f3d9c037d35 - rustc_query_system[f9705d2eaa4cb9d4]::query::plumbing::try_execute_query::<rustc_query_impl[38c8fda5da4ce4ba]::DynamicConfig<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_span[15bad65b23151bca]::def_id::DefId, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[38c8fda5da4ce4ba]::plumbing::QueryCtxt, false>
  27:     0x7f3d9d8e363f - rustc_query_impl[38c8fda5da4ce4ba]::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
  28:     0x7f3d9c63eac0 - rustc_middle[ad940b573bf0e24a]::query::plumbing::query_get_at::<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_span[15bad65b23151bca]::def_id::DefId, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>>
  29:     0x7f3d9c622f0b - <rustc_trait_selection[a0a161ca2aebd8f0]::traits::project::AssocTypeNormalizer as rustc_type_ir[60cd2614b586a6cd]::fold::TypeFolder<rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt>>::fold_ty
  30:     0x7f3d9c630dff - rustc_trait_selection[a0a161ca2aebd8f0]::traits::project::opt_normalize_projection_type
  31:     0x7f3d9cefe34d - <rustc_infer[57bce297a81275c8]::infer::InferCtxtBuilder as rustc_trait_selection[a0a161ca2aebd8f0]::infer::InferCtxtBuilderExt>::enter_canonical_trait_query::<rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::sty::AliasTy>, rustc_middle[ad940b573bf0e24a]::traits::query::NormalizationResult, rustc_traits[cc8add84767b24b0]::normalize_projection_ty::normalize_projection_ty::{closure#0}>
  32:     0x7f3d9c0a28a7 - rustc_query_impl[38c8fda5da4ce4ba]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[38c8fda5da4ce4ba]::query_impl::normalize_projection_ty::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>
  33:     0x7f3d9d30c5e1 - rustc_query_system[f9705d2eaa4cb9d4]::query::plumbing::try_execute_query::<rustc_query_impl[38c8fda5da4ce4ba]::DynamicConfig<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_middle[ad940b573bf0e24a]::infer::canonical::Canonical<rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::sty::AliasTy>>, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[38c8fda5da4ce4ba]::plumbing::QueryCtxt, false>
  34:     0x7f3d9d30c1e1 - rustc_query_impl[38c8fda5da4ce4ba]::query_impl::normalize_projection_ty::get_query_non_incr::__rust_end_short_backtrace
  35:     0x7f3d9cc299e6 - <rustc_trait_selection[a0a161ca2aebd8f0]::traits::query::normalize::QueryNormalizer as rustc_type_ir[60cd2614b586a6cd]::fold::FallibleTypeFolder<rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt>>::try_fold_ty
  36:     0x7f3d9cf3caf1 - <rustc_traits[cc8add84767b24b0]::normalize_erasing_regions::provide::{closure#0} as core[cd6e598b5651e7ef]::ops::function::FnOnce<(rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt, rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::generic_args::GenericArg>)>>::call_once
  37:     0x7f3d9c01ef56 - rustc_query_impl[38c8fda5da4ce4ba]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[38c8fda5da4ce4ba]::query_impl::try_normalize_generic_arg_after_erasing_regions::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>
  38:     0x7f3d9cfd831b - rustc_query_system[f9705d2eaa4cb9d4]::query::plumbing::try_execute_query::<rustc_query_impl[38c8fda5da4ce4ba]::DynamicConfig<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::generic_args::GenericArg>, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[38c8fda5da4ce4ba]::plumbing::QueryCtxt, false>
  39:     0x7f3d9cfd7ff4 - rustc_query_impl[38c8fda5da4ce4ba]::query_impl::try_normalize_generic_arg_after_erasing_regions::get_query_non_incr::__rust_end_short_backtrace
  40:     0x7f3d9c5b6329 - <rustc_middle[ad940b573bf0e24a]::ty::normalize_erasing_regions::TryNormalizeAfterErasingRegionsFolder as rustc_type_ir[60cd2614b586a6cd]::fold::FallibleTypeFolder<rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt>>::try_fold_ty
  41:     0x7f3d9c8686f6 - rustc_ty_utils[f751fc2bbef1180a]::layout::layout_of
  42:     0x7f3d9bff98ee - rustc_query_impl[38c8fda5da4ce4ba]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[38c8fda5da4ce4ba]::query_impl::layout_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 16usize]>>
  43:     0x7f3d9c85cd41 - rustc_query_system[f9705d2eaa4cb9d4]::query::plumbing::try_execute_query::<rustc_query_impl[38c8fda5da4ce4ba]::DynamicConfig<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::Ty>, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 16usize]>>, false, true, false>, rustc_query_impl[38c8fda5da4ce4ba]::plumbing::QueryCtxt, false>
  44:     0x7f3d9c85c943 - rustc_query_impl[38c8fda5da4ce4ba]::query_impl::layout_of::get_query_non_incr::__rust_end_short_backtrace
  45:     0x55f86c514976 - rustc_middle[ad940b573bf0e24a]::query::plumbing::query_get_at::<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::DefaultCache<rustc_middle[ad940b573bf0e24a]::ty::ParamEnvAnd<rustc_middle[ad940b573bf0e24a]::ty::Ty>, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 16usize]>>>
  46:     0x55f86c51f530 - clippy_utils[500093ed3c1879d3]::ty::approx_ty_size
  47:     0x55f86c36d0b9 - clippy_lints[ba76610b7f8d4acd]::functions::result::check_result_large_err
  48:     0x55f86c36cf5f - clippy_lints[ba76610b7f8d4acd]::functions::result::check_trait_item
  49:     0x55f86c3166fe - <clippy_lints[ba76610b7f8d4acd]::functions::Functions as rustc_lint[60dc77c1de2b1a7a]::passes::LateLintPass>::check_trait_item
  50:     0x7f3d9e3e6226 - <rustc_lint[60dc77c1de2b1a7a]::late::RuntimeCombinedLateLintPass as rustc_lint[60dc77c1de2b1a7a]::passes::LateLintPass>::check_trait_item
  51:     0x7f3d9e35b24f - <rustc_lint[60dc77c1de2b1a7a]::late::LateContextAndPass<rustc_lint[60dc77c1de2b1a7a]::late::RuntimeCombinedLateLintPass> as rustc_hir[f915117a566b7aff]::intravisit::Visitor>::visit_nested_trait_item
  52:     0x7f3d9e3561c5 - rustc_hir[f915117a566b7aff]::intravisit::walk_item::<rustc_lint[60dc77c1de2b1a7a]::late::LateContextAndPass<rustc_lint[60dc77c1de2b1a7a]::late::RuntimeCombinedLateLintPass>>
  53:     0x7f3d9e35ad8f - <rustc_lint[60dc77c1de2b1a7a]::late::LateContextAndPass<rustc_lint[60dc77c1de2b1a7a]::late::RuntimeCombinedLateLintPass> as rustc_hir[f915117a566b7aff]::intravisit::Visitor>::visit_nested_item
  54:     0x7f3d9e355788 - rustc_hir[f915117a566b7aff]::intravisit::walk_mod::<rustc_lint[60dc77c1de2b1a7a]::late::LateContextAndPass<rustc_lint[60dc77c1de2b1a7a]::late::RuntimeCombinedLateLintPass>>
  55:     0x7f3d9d5827c9 - rustc_lint[60dc77c1de2b1a7a]::late::late_lint_crate::<rustc_lint[60dc77c1de2b1a7a]::BuiltinCombinedLateLintPass>
  56:     0x7f3d9d581c78 - <rustc_session[1cfb20cc2da267b8]::session::Session>::time::<(), rustc_lint[60dc77c1de2b1a7a]::late::check_crate<rustc_lint[60dc77c1de2b1a7a]::BuiltinCombinedLateLintPass, rustc_interface[fb497e4e8d0d6e95]::passes::analysis::{closure#5}::{closure#1}::{closure#2}::{closure#0}::{closure#0}>::{closure#0}::{closure#0}>
  57:     0x7f3d9d516df3 - <rustc_session[1cfb20cc2da267b8]::session::Session>::time::<(), rustc_interface[fb497e4e8d0d6e95]::passes::analysis::{closure#5}::{closure#1}::{closure#2}::{closure#0}>
  58:     0x7f3d9d51696a - <core[cd6e598b5651e7ef]::panic::unwind_safe::AssertUnwindSafe<rustc_interface[fb497e4e8d0d6e95]::passes::analysis::{closure#5}::{closure#1}> as core[cd6e598b5651e7ef]::ops::function::FnOnce<()>>::call_once
  59:     0x7f3d9d41b178 - <rustc_session[1cfb20cc2da267b8]::session::Session>::time::<(), rustc_interface[fb497e4e8d0d6e95]::passes::analysis::{closure#5}>
  60:     0x7f3d9d419ea7 - rustc_interface[fb497e4e8d0d6e95]::passes::analysis
  61:     0x7f3d9d48c76a - rustc_query_impl[38c8fda5da4ce4ba]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[38c8fda5da4ce4ba]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 1usize]>>
  62:     0x7f3d9d48c759 - <rustc_query_impl[38c8fda5da4ce4ba]::query_impl::analysis::dynamic_query::{closure#2} as core[cd6e598b5651e7ef]::ops::function::FnOnce<(rustc_middle[ad940b573bf0e24a]::ty::context::TyCtxt, ())>>::call_once
  63:     0x7f3d9d6225c8 - rustc_query_system[f9705d2eaa4cb9d4]::query::plumbing::try_execute_query::<rustc_query_impl[38c8fda5da4ce4ba]::DynamicConfig<rustc_query_system[f9705d2eaa4cb9d4]::query::caches::SingleCache<rustc_middle[ad940b573bf0e24a]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[38c8fda5da4ce4ba]::plumbing::QueryCtxt, false>
  64:     0x7f3d9d622357 - rustc_query_impl[38c8fda5da4ce4ba]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  65:     0x7f3d9d1b34a5 - <rustc_middle[ad940b573bf0e24a]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[7a6d5a32edd3e2a]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>>
  66:     0x7f3d9d1b22e9 - rustc_span[15bad65b23151bca]::set_source_map::<core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>, rustc_interface[fb497e4e8d0d6e95]::interface::run_compiler<core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>, rustc_driver_impl[7a6d5a32edd3e2a]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  67:     0x7f3d9d1ace37 - std[28b408c5a4d1cfeb]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[fb497e4e8d0d6e95]::util::run_in_thread_pool_with_globals<rustc_interface[fb497e4e8d0d6e95]::interface::run_compiler<core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>, rustc_driver_impl[7a6d5a32edd3e2a]::run_compiler::{closure#1}>::{closure#0}, core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>>
  68:     0x7f3d9d1ac76e - <<std[28b408c5a4d1cfeb]::thread::Builder>::spawn_unchecked_<rustc_interface[fb497e4e8d0d6e95]::util::run_in_thread_pool_with_globals<rustc_interface[fb497e4e8d0d6e95]::interface::run_compiler<core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>, rustc_driver_impl[7a6d5a32edd3e2a]::run_compiler::{closure#1}>::{closure#0}, core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[cd6e598b5651e7ef]::result::Result<(), rustc_span[15bad65b23151bca]::ErrorGuaranteed>>::{closure#1} as core[cd6e598b5651e7ef]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  69:     0x7f3d9ad63c75 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h388b23a3322b34b3
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/alloc/src/boxed.rs:2007:9
  70:     0x7f3d9ad63c75 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h52bd1f089925e77e
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/alloc/src/boxed.rs:2007:9
  71:     0x7f3d9ad63c75 - std::sys::unix::thread::Thread::new::thread_start::h422b2dcbc6cfda3a
                               at /rustc/399b068235ceea440540539b3bfd1aeb82214a28/library/std/src/sys/unix/thread.rs:108:17
  72:     0x7f3d9ac1e609 - start_thread
  73:     0x7f3d9a9f2133 - clone
  74:                0x0 - <unknown>
  ```

  </p>
</details>

**Comment by PhiSyX** [comment]
This error has been corrected in new versions.

```
rustc 1.73.0-nightly (474709a9a 2023-08-03)
binary: rustc
commit-hash : 474709a9a2a74a8bcf0055fadb335d0ca0d2d939
commit-date: 2023-08-03
host: x86_64-pc-windows-msvc
release: 1.73.0-nightly
LLVM version: 16.0.5
clippy 0.1.73 (474709a9 2023-08-03)
```
